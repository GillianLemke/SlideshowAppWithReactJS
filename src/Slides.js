import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import Title from './title';
import Simple from './simple';
import TwoColumn from './twocolumn';
import $ from 'jquery'

export default class Slides extends React.Component {
  constructor() {
    super();
    this.state = {
      type: '',
    };
  }

  // pass url in as param?
  getTypeFromAPI() {
    return $.getJSON('google.com/type')
      .then((data) => {
        this.setState({ type:data.results })
      });
  }

  getType() {
    // is using var okay? const or let instead?
    this.getTypeFromAPI();
    var slideType = this.state.type;

    // are these types named like this?
    if (slideType === 'title') {
      return (<Title />);
    } else if (slideType === 'simple') {
      return (<Simple />);
    } else if (slideType === 'twocolumn') {
      return (<TwoColumn />);
    } else {
      return (<div></div>);
    }
  }

  render() {
    return(<div>{this.getType()}</div>);
  }

}
