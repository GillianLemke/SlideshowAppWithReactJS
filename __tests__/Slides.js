import React from 'react'
import Slides from '../src/Slides.js'
import {shallow} from 'enzyme' // could also mount
import toJson from 'enzyme-to-json'

describe('Slides', () => {
  it('renders', () => {
    const wrapper = shallow(<Slides type='' />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  it('returns a title slide', () => {
    const wrapper = shallow(<Slides type='title' />)
    // can a method be used here?
    expect(wrapper.find('title-slide').length) >= 1
    expect(wrapper.find('simple-slide').length)===0
    expect(wrapper.find('two-column-slide').length)===0
  });
  it('returns a simple slide', () => {
    const wrapper = shallow(<Slides type='simple' />)
    expect(wrapper.find('title-slide').length)===0
    expect(wrapper.find('simple-slide').length)>=1
    expect(wrapper.find('two-column-slide').length)===0
  });
  it('returns a two column slide'), () => {
    const wrapper = shallow(<Slides type='twocolumn' />)
    expect(wrapper.find('title-slide').length)===0
    expect(wrapper.find('simple-slide').length)===0
    expect(wrapper.find('two-column-slide').length)>=1
  }


});
