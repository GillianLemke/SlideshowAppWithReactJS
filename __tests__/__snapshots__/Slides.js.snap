exports[`Slides renders 1`] = `
<div>
  <div />
</div>
`;
